### PR TITLE
move swagger-ui to webpack/npm and update it to 3.24.3

### PR DESCRIPTION
**Pull request by silverwind**
Created a second webpack output file for swagger-ui which is loaded on the /api/swagger route. One notable difference is the absence of the swagger favicon that was previously used which is now the gitea icon. I see no easy way to restore that favicon, so I decided to not keep it.

### CODE REVIEW

**Comment by silverwind** [comment]
Also, this will update `swagger-ui` from 3.13.4 to 3.24.3.

**Comment by 6543** [comment]
@silverwind can you add a screenshot?

**Comment by silverwind** [comment]
Screenshot added, but it's kind of pointless because page content has not changed 😉

**Comment by 6543** [comment]
thought it would change more ... :sweat_smile:  - code change looks good and local test worked :)

**Review by 6543 — APPROVED** [review]

**Comment by sapk** [comment]
@silverwind doesn't this mean that the swagger.js will be build each time ? 
It feel better to keep this outside from the gitea js config.

For info, by updating swagger-ui this will fix few security issues (but maybe not need a backport since only xss that we shouldn't be affected)
- https://github.com/swagger-api/swagger-ui/releases/tag/v3.18.0
- https://github.com/swagger-api/swagger-ui/releases/tag/v3.23.11
- https://github.com/swagger-api/swagger-ui/releases/tag/v3.24.1

File: webpack.config.js
Comment (sapk) [inline webpack.config.js]: why disabling performance hint ?

Comment (silverwind) [reply webpack.config.js]: To prevent these warnings triggered by the swagger bundle:

```
WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  swagger.js (1.29 MiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  swagger (1.29 MiB)
      swagger.js

**Comment by silverwind** [comment]
> doesn't this mean that the swagger.js will be build each time ?

It will only build once per change in the `web_src` directory, or when `node_modules` change.

Not 100% sure if `make` is smart enough to track changes in `node_modules`, so it might be better to track `package-lock.json` instead.

File: webpack.config.js
Comment (sapk) [reply webpack.config.js]: Disabling for the whole project and also for gitea js seems bad. Maybe filter swagger with [assetFilter](https://webpack.js.org/configuration/performance/#performanceassetfilter)

**Comment by silverwind** [comment]
>  It feel better to keep this outside from the gitea js config.

I think it rightly belongs in the JS parts and swagger-ui also [recommends](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/installation.md#npm-registry) the bunling approach over vendoring.

File: webpack.config.js
Comment (silverwind) [reply webpack.config.js]: Done

Comment (silverwind) [reply webpack.config.js]: For the record, on another run, I also saw warnings for the source map files, so I added the filter on the `.js` file extension:

```
WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  index.js.map (455 KiB)
  swagger.js.map (4.61 MiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  index (583 KiB)
      index.js
      index.js.map
  swagger (4.61 MiB)
      swagger.js.map

File: webpack.config.js
Comment (sapk) [inline webpack.config.js:72]: Better use the default with a special case.
`return !((/\.map$/.test(filename)) || (filename === 'swagger.js'))`

**Review by sapk — APPROVED** [review]
I have doubt if this should be tightly configured to gitea js build but this is an improvement and it could be improved again later if it cause problem.

**Review by techknowlogick — DISMISSED** [review]
Thanks for PR, I’m on mobile and am having difficulty adding a comment so I am blocking this until I can get to a desktop to add review.

File: webpack.config.js
Comment (silverwind) [reply webpack.config.js:72]: Ok, committed a slight variation of this.

**Comment by silverwind** [comment]
Investigating removal of source map for `swagger.js` as it's rather pointless for something that is 99.99% third party code.

File: templates/swagger/ui.tmpl
Comment (techknowlogick) [inline templates/swagger/ui.tmpl]: Finally at a computer, and can add a few notes.
```suggestion
<script src="{{StaticUrlPrefix}}/js/swagger.js?v={{MD5 AppVer}}"> </script>
```

 and can you also add this js file to the service worker too?

**Comment by silverwind** [comment]
@sapk I switched to `swagger-ui-dist`, let me know what you think. Now, it just copies the files to `public` during the webpack build. That way, build will be faster with the downside that we can no longer control the polyfills for the swagger-ui code.

**Comment by silverwind** [comment]
Let's wait on @sapk to approve the latest change to `swagger-ui-dist` before merging. I did it based on request from https://github.com/go-gitea/gitea/pull/9715#issuecomment-573367599. Personally, I have a slight preference for the previous `swagger-ui` variant because it allows us to control the polyfills.

**Comment by silverwind** [comment]
Decided to go with `swagger-ui` approach. It just seems cleaner that way. The webpack build completes in 6 seconds on my machine with `swagger-ui` included in the build, which seems plenty fast to me.

Good to go from my side.

**Comment by sapk** [comment]
@silverwind It is more the fact that it is in the same build process than gitea js that bother me the most compared to using directly release files but it feel more a personal intuition and not backed by real experience. So I am totally ok with merging this and see if it become a problem (which will be limited since only for builder) or if I was wrong to think so. Otherwise this PR totally LGTM.

**Comment by silverwind** [comment]
Okay. When it comes to JS modules, I don't expect release files to be present and many modules do not even provide them because it is generally expected that the user takes care of their bundling/polyfilling needs. `swagger-ui` actually goes the extra mile to provide those files separately, which is nice, but unnecessary given that we already bundle.

**Comment by sapk** [comment]
By release file, I mean like previously, totally outside npm/node_modules but don't bother with that.

**Comment by lafriks** [comment]
I would also prefer that we build dependencies ourselves as most libraries in npm are provided with idea to be built on end product

**Comment by silverwind** [comment]
Squashed and pushed a few minor cosmetic tweaks.

**Comment by silverwind** [comment]
CI failed with unrelated error, please restart.

````
level=error msg="Running error: ctrlflow: failed prerequisites: inspect@code.gitea.io/gitea/modules/markup/markdown [code.gitea.io/gitea/modules/markup.test]"
````

**Comment by silverwind** [comment]
Rebased. @techknowlogick want to lift your block?

**Pull request merged**
